q.cpp

```diff
@@ -0,0 +1,245 @@
+#include <stdio.h>
+#include <string.h>
+#include <malloc.h>
```

Comment:
It's simple.   make uses default rules.
You don't even need a makefile to "make q" :-)